### DISCUSSION TITLE
Please add "How to add xmake to existing project?" to the FAQ

### DISCUSSION THREAD

**Discussion by xparq** [xmake-io/xmake#3369] [Q&A]
### Is your feature request related to a problem? Please describe.

Yes, the problem is that I got stuck right at the beginning of a bold attempt to just go ahead -- with the possible help of the Quick Start guide & the FAQ -- and integrate xmake into my existing prj. with a hand-crafted Makefile, which _I'd like to replace, rather than be reused, by xmake!_

But I found nothing about that after a _quick_ glance at the general docs + a _full_ read of the FAQ.
Also, just running `xmake` for the prj dir didn't offer to start migrating from my legacy NMAKE setup (despite there's not much point in using `xmake` just to the same as I've been doing by typing `nmake` instead).

### Describe the solution you'd like

Get explicit help from the primary docs. for this (supposedly very common) migration use case.

### Describe alternatives you've considered

Another build tools, like build2.

### Additional context

Nice project BTW! :)

**Comment by waruqi** [comment]
The documentation does not provide a detailed description of how to migrate from other makefile/cmakelists files to xmake.lua, because the makefile/cmakelists of each third-party project are very complex and there is no common migration solution.

You will need to understand the basic configuration and API usage of xmake.lua. Then understand the configuration of the makefile before you migrate it.

like this

```lua
target("test")
    set_kind("binary")
    add_files("src/*.c")
    add_defines("TEST")
```

If there are extra cflags and macro definitions in the makefile, you can set them with add_cflags or add_defines, you can read the xmake API manual first.

https://xmake.io/#/manual/project_target

**Comment by lewisl** [comment]
I think he explicitly was saying DO NOT convert my existing makefile.  Instead he was saying: "I am NOT starting from scratch." I have folders of code already.  I want the top level folder of the grouping of code folders (my programming language "project") to be an xmake project. in other words, add the xmake init stuff within an existing folder that contains code already. Every tool maker thinks that their tool is the most important and will be the first to bootstrap a project. But, that's not true.  Probably the IDE's project comes first and then the compiler's project is second and Xmake's definition of project comes third. You don't document how to add on or inject or instantiate an xmake project where something exists.  You can make all the xmake folders you want in the .xmake and .build directories.  Just leave the source code folders as they are.  How can this be done?

**Comment by lewisl** [comment]
You sort of can by creating and configuring.  It should be more obvious.  Adding Xmake last to a source folder is probably a much more common work habit than doing it first. The startup project isn't that helpful.  IDE's try harder at that and even that can be annoying.  Xmake is awesome; love it.  Freedom fits its vibe!
